filewatcher/filewatcher_test.go

```diff
@@ -79,16 +98,7 @@ func TestFileWatcher(t *testing.T) {
 	defer data.Stop()
 	compareBytesData(t, data.Get(), payload1)
 
-	func() {
-		f, err := os.OpenFile(path, os.O_WRONLY|os.O_TRUNC, 0644)
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer f.Close()
-		if _, err := f.Write(payload2); err != nil {
-			t.Fatal(err)
-		}
-	}()
+	overwriteFile(t, path, bytes.NewReader(payload2))
```

Comment:
🔕 you use bytes.NewReader for both, so you might as well just pass the contents as `[]byte` and use `os.WriteFile`

Reply:
Looking at the source code of `os.WriteFile`, it tries to do a single `Write`, which I don't really like and it's more likely to fail if we are trying to write large content (although we don't write large content here). While `io.Copy` handles those cases much more thoroughly.

Reply:
You can refactor `overwriteFile` in the future _if_ you end up with a reader.  YAGNI / KISS / premature optimization / etc.

Reply:
Changed to use `os.WriteFile`. Looks like `os.File.Write` itself implemented big write handling, at least for linux.